src/registrar/utility/csv_export.py

```diff
@@ -182,8 +184,13 @@ def write_csv(
                 # It indicates that DomainInformation.domain is None.
                 logger.error("csv_export -> Error when parsing row, domain was None")
                 continue
+        total_body_rows.extend(rows)
```

Comment:
**Question:** To confirm -- we're also getting the row count and extending it once versus writing the row each time here, correct?

Reply:
Yeah. That's not a good thing. That's needed because we can't write body rows before we write the header. We can't write the header before we know max-dm. We can't know max_dm before we iterate over all DIs. We get the row data while iterating over DIs. So, we hold off on writing and store them all in a mega-list until the header is done.

Reply:
So there is a difference on writre row vs rows but it's dependent on the amount of data we have. Memory wise, yes storing all rows in memory is costly on memory but not on computation time. Whereas as writerow on large data sets is more expensive in time but less expensive in memory. Time is kind of our killer here and is really dependent on the large data set. A timer on the methods would require a large data set to get an accurate view, but the documentation online on row vs rows is a bit more enlightening. Writerows seems like the better call based on online searches